pkg/operator/sync_test.go

```diff
@@ -118,3 +123,246 @@ func TestWaitForDeploymentRollout(t *testing.T) {
 		})
 	}
 }
+
+func TestSyncValidatingWebhooks(t *testing.T) {
+	defaultHook := mapiv1.NewValidatingWebhookConfiguration()
+	referenceHook := mapiv1.NewValidatingWebhookConfiguration()
+	referenceHook.SetName("test")
+
+	testCases := []struct {
+		testCase       string
+		exisingWebhook func() *admissionregistrationv1.ValidatingWebhookConfiguration
+		expectUpdate   bool
+	}{
+		{
+			testCase:     "It should create webhook if it does not exsit",
+			expectUpdate: true,
+		},
+		{
+			testCase:     "It should update webhookConfiguration if some of their webhooks differ",
+			expectUpdate: true,
+			exisingWebhook: func() *admissionregistrationv1.ValidatingWebhookConfiguration {
+				hook := defaultHook.DeepCopy()
```

Comment:
I wish we use the same terms consistently across the whole code i.e there we don't need more than 2 different terms: `webhookConfiguration` and `webhook`